icepyx/tests/test_quest.py

```diff
@@ -40,9 +40,9 @@ def test_add_argo(quest_instance):
 
     obs = quest_instance.datasets
 
-    assert type(obs) == dict
+    assert isinstance(obs, dict)
     assert exp_key in obs.keys()
-    assert type(obs[exp_key]) == exp_type
+    assert isinstance(obs[exp_key], exp_type)
```

Comment:
Note: this is a test so we have automatic confirmation it's fine, but for anyone not aware, there is a difference in behavior between `type(...) == ...` and `isintance(..., ...)` - `isinstance` evaluates `True` with inheritance:


```python
>>> class Foo(str):
...     ...
... 
>>> type(Foo()) is str
False
>>> type(Foo()) is Foo
True
>>> isinstance(Foo(), str)
True
>>> isinstance(Foo(), Foo)
True
```

Reply:
Agreed.  The `==` --> `is` transition would be good enough to silence this ruff rule.  The "comparing types" thing is frowned on in PEP8 and there is a different ruff rule that covers that.

Reply:
The more I think about it, the more I think we should go back to exact type comparison with `type(...) is ...`. Since this is a test, loosening the comparison enables the underlying code to change more without breaking the test. We an always come back to this. My little mind is just thinking about the "foolish consistency" part of PEP8.